src/metrics/abc.rs

```diff
@@ -0,0 +1,803 @@
+            STAREQ | SLASHEQ | PERCENTEQ | DASHEQ | PLUSEQ | LTLTEQ | GTGTEQ | AMPEQ | PIPEEQ
+            | CARETEQ | GTGTGTEQ | PLUSPLUS | DASHDASH => {
+                stats.assignments += 1.;
+            }
+            FieldDeclaration | LocalVariableDeclaration => {
+                stats.declaration.push(DeclKind::Var);
+            }
+            Final => {
+                if let Some(DeclKind::Var) = stats.declaration.last() {
+                    stats.declaration.push(DeclKind::Const);
+                }
+            }
+            SEMI => {
+                if let Some(DeclKind::Const | DeclKind::Var) = stats.declaration.last() {
+                    stats.declaration.clear();
+                }
+            }
+            EQ => {
+                // Excludes constant declarations
+                if let Some(decl) = stats.declaration.last() {
+                    if let DeclKind::Var = decl {
+                        stats.assignments += 1.;
+                    }
+                } else {
+                    stats.assignments += 1.;
+                }
+            }
+            MethodInvocation | New => {
+                stats.branches += 1.;
+            }
+            GTEQ | LTEQ | EQEQ | BANGEQ | Else | Case | Default | QMARK | Try | Catch => {
+                stats.conditions += 1.;
+            }
+            GT | LT => {
+                // Ignores < and > used for generic types
+                if let Some(parent) = node.object().parent() {
+                    if !matches!(parent.kind_id().into(), TypeArguments) {
+                        stats.conditions += 1.;
+                    }
+                }
+            }
+            // Counts unary conditions of elements separated by boolean operators
+            AMPAMP | PIPEPIPE => {
+                if let Some(parent) = node.object().parent() {
+                    if let Some((res, _)) = count_java_unary_conditions(&Node::new(parent)) {
+                        stats.conditions += res;
+                    }
+                }
+            }
+            // Counts unary conditions inside if, while and do-while statements
+            ParenthesizedExpression => {
+                if let Some(parent) = node.object().parent() {
+                    match parent.kind_id().into() {
+                        IfStatement | WhileStatement | DoStatement => {
+                            if let Some((res, _)) = count_java_unary_conditions(node) {
+                                stats.conditions += res;
+                            }
+                        }
+                        _ => {}
+                    }
+                }
+            }
+            // Counts unary conditions inside for statements
+            ForStatement => {
+                // Position of the condition if the for counter has already been declared outside the for
+                // e.g. `for ( i=0; ... ; ... ) {}`
+                if let Some(condition) = node.object().child(3) {
```

Comment:
Perhaps we can explicitly state that `3` is related to the comment above saying: *3 magic number represents this situation...*